/*
 *   This file is part of Nudnik. <https://github.com/salosh/nudnik.git>
 *
 *   Nudnik is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   Nudnik is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with Nudnik.  If not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto3";

package nudnik.entity;

service Parser{
  rpc parse(Request) returns (Response) {}
}

message Load {
  enum LoadType {
    rtt  = 0;
    rttr = 1;
    cpu  = 2;
    mem  = 3;
    bcmd = 4;
    fcmd = 5;
  }
  LoadType load_type = 1;
  string value = 2;
}

message Request {
  string name        = 1;
  int32  stream_id   = 2;
  int32  worker_id   = 3;
  int32  sequence_id = 4;
  int32  message_id  = 5;
  int64  ctime       = 6;
  int64  stime       = 7;
  int64  rtime       = 8;
  int32  rcount      = 9;
  bytes  meta        = 10;
  repeated Load load = 11;
}

message Response {
  enum StatusCode {
    OK = 0;
    SERVER_ERROR = 1;
    CLIENT_ERROR = 2;
  }
  StatusCode status_code = 1;
  int64 ctime            = 2;
  int64 ltime            = 3;
  int64 stime            = 4;
  bytes  meta            = 5;
}

